api/apiclient.go

```diff
@@ -738,14 +845,13 @@ func (d dialer) dial1() (jsoncodec.JSONConn, *tls.Config, error) {
 	}
 	// It's possible we're inappropriately using the private
 	// CA certificate, so retry immediately with the public one.
-	tlsConfig1 := *d.opts.tlsConfig
-	tlsConfig1.RootCAs = nil
-	tlsConfig1.ServerName = d.opts.sniHostName
-	conn, err = d.opts.DialWebsocket(d.ctx, d.urlStr, &tlsConfig1)
+	tlsConfig.RootCAs = nil
+	tlsConfig.ServerName = d.serverName
+	conn, err = d.opts.DialWebsocket(d.ctx, d.urlStr, tlsConfig, d.ipAddr)
```

Comment:
This will conflict with the patch that Witold just landed to always return the original error, instead of sometimes returning the fallback path.